services/director-v2/src/simcore_service_director_v2/models/schemas/dynamic_services/monitor.py

```diff
@@ -0,0 +1,487 @@
+    dy-proxy_4dde07ea-73be-4c44-845a-89479d1556cf
+
+    dynamic sidecar structure
+    0. a network is created: dy-sidecar_4dde07ea-73be-4c44-845a-89479d1556cf
+    1. a dynamic-sidecar is started: dy-sidecar_4dde07ea-73be-4c44-845a-89479d1556cf
+    a traefik instance: dy-proxy_4dde07ea-73be-4c44-845a-89479d1556cf
+    """
+
+    service_name_dynamic_sidecar: str = Field(
+        ...,
+        regex=REGEX_DY_SERVICE_SIDECAR,
+        max_length=MAX_ALLOWED_SERVICE_NAME_LENGTH,
+        description="unique name of the dynamic-sidecar service",
+    )
+    proxy_service_name: str = Field(
+        ...,
+        regex=REGEX_DY_SERVICE_PROXY,
+        max_length=MAX_ALLOWED_SERVICE_NAME_LENGTH,
+        description="name of the proxy for the dynamic-sidecar",
+    )
+
+    simcore_traefik_zone: str = Field(
+        ...,
+        regex=REGEX_DY_SERVICE_SIDECAR,
+        description="unique name for the traefik constraints",
+    )
+    dynamic_sidecar_network_name: str = Field(
+        ...,
+        regex=REGEX_DY_SERVICE_SIDECAR,
+        description="based on the node_id and project_id",
+    )
+
+    @classmethod
+    def make(cls, node_uuid: UUID) -> "DynamicSidecarNames":
+        return cls(
+            service_name_dynamic_sidecar=assemble_service_name(
+                DYNAMIC_SIDECAR_SERVICE_PREFIX, node_uuid
+            ),
+            proxy_service_name=assemble_service_name(
+                DYNAMIC_PROXY_SERVICE_PREFIX, node_uuid
+            ),
+            simcore_traefik_zone=f"{DYNAMIC_SIDECAR_SERVICE_PREFIX}_{node_uuid}",
+            dynamic_sidecar_network_name=f"{DYNAMIC_SIDECAR_SERVICE_PREFIX}_{node_uuid}",
+        )
+
+
+class MonitorData(BaseModel):
+    service_name: str = Field(
+        ..., description="Name of the current dynamic-sidecar being monitored"
+    )
+
+    node_uuid: NodeID = Field(
+        ..., description="the node_id of the service as defined in the workbench"
+    )
+
+    project_id: ProjectID = Field(
+        ..., description="project_uuid required by the status"
+    )
+
+    user_id: UserID = Field(..., description="user_id required by the status")
+
+    dynamic_sidecar: DynamicSidecar = Field(
+        ...,
+        description="stores information fetched from the dynamic-sidecar",
+    )
+
+    service_key: str = Field(
+        ...,
+        regex=SERVICE_KEY_RE,
+        description="together with the tag used to compose the docker-compose spec for the service",
+    )
+    service_tag: str = Field(
+        ...,
+        regex=VERSION_RE,
+        description="together with the key used to compose the docker-compose spec for the service",
+    )
+    paths_mapping: PathsMappingLabel = Field(
+        ...,
+        description=(
+            "the service explicitly requests where to mount all paths "
+            "which will be handeled by the dynamic-sidecar"
+        ),
+    )
+    compose_spec: ComposeSpecLabel = Field(
+        ...,
```

Comment:
> this seems to have a lot of the entries already defined in ServiceSettingsLabels.
> would it not be simpler, less error prone (reducing duplication) and easier to understand to actually compose that model using the others? it would also reduce the number of lines by a lot.


Yes 3/4 entries are shared with `ServiceSettingsLabels`, but in this model all 3 of them are required, which is a different from what is defined in `ServiceSettingsLabels`. So this one will not get changed.

Reply:
So I've removed some of these properties and inherited from a common model.